### PR TITLE
Register amqp/worker commands as services

### PR DESCRIPTION
**Pull request by chalasr**
- [x] Moved commands to their respective components
- [x] Registered them as services in the framework instead of container aware
- [x] Added more output
- [x] Improved output style
- [x] Replaced parameters by direct injection
- [x] Made use of `Command::setProcessTitle()`

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/amqp.xml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        <defaults public="false" />
+
+        <service id="amqp.broker_locator" class="Symfony\Component\DependencyInjection\ServiceLocator">
+            <tag name="container.service_locator" />
+            <argument type="collection">
+                <argument key="Symfony\Component\Amqp\Broker" type="service" id="amqp.broker" />
+            </argument>
+        </service>
+
+        <service id="worker.command.list" class="Symfony\Component\Worker\Command\WorkerListCommand">
```
Comment (lyrixx) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/amqp.xml]: This service is already defined in the worker.xml file

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/worker.xml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        </service>
+
+        <service id="worker.command.list" class="Symfony\Component\Worker\Command\WorkerListCommand">
+            <tag name="console.command" />
+            <argument /> <!-- worker names -->
+        </service>
+
+        <service id="worker.command.run" class="Symfony\Component\Worker\Command\WorkerRunCommand">
+            <tag name="console.command" />
+            <argument type="service" id="worker.worker_locator" /> <!-- worker locator -->
```
Comment (lyrixx) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/worker.xml]: is this comment useful?

File: src/Symfony/Component/Worker/Command/WorkerListCommand.php [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+            ->setDescription('Lists available workers.')
+        ;
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $io = new SymfonyStyle($input, $output);
+
+        if (!$this->workers) {
+            $io->getErrorStyle()->error('There is no available worker.');
```
Comment (lyrixx) [inline src/Symfony/Component/Worker/Command/WorkerListCommand.php]: `there are no available workers`

File: src/Symfony/Component/Worker/Command/WorkerListCommand.php [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+            return;
+        }
+
+        $io->getErrorStyle()->title('Available workers');
```
Comment (lyrixx) [inline src/Symfony/Component/Worker/Command/WorkerListCommand.php]: why did you get the **error**style?

File: src/Symfony/Component/Worker/Command/WorkerRunCommand.php [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+            ->setDefinition(array(
+                new InputArgument('worker', InputArgument::REQUIRED, 'The worker'),
+                new InputOption('name', null, InputOption::VALUE_REQUIRED, 'A name, useful for stats/monitoring. Defaults to worker name.'),
+            ))
+        ;
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $workerName = $input->getArgument('worker');
```
Comment (lyrixx) [inline src/Symfony/Component/Worker/Command/WorkerRunCommand.php]: why did you remove the check on the availability of the pcntl extension ?

File: src/Symfony/Component/Worker/Command/WorkerListCommand.php (hunk 4 above)
Comment (chalasr) [inline src/Symfony/Component/Worker/Command/WorkerListCommand.php]: Titles are not useful to redirect as stdout, this makes use of stderr

File: src/Symfony/Component/Worker/Command/WorkerRunCommand.php (hunk 5 above)
Comment (chalasr) [inline src/Symfony/Component/Worker/Command/WorkerRunCommand.php]: because the command has been moved in Worker which requires `ext-pcntl`, this code can't be reached without pcntl enabled

Comment (lyrixx) [inline src/Symfony/Component/Worker/Command/WorkerRunCommand.php]: If I required symfony/symfony (and not symfony/worker), it does.

File: src/Symfony/Component/Worker/Command/WorkerListCommand.php (hunk 4 above)
Comment (lyrixx) [inline src/Symfony/Component/Worker/Command/WorkerListCommand.php]: :+1:

File: src/Symfony/Component/Worker/Command/WorkerRunCommand.php (hunk 5 above)
Comment (chalasr) [inline src/Symfony/Component/Worker/Command/WorkerRunCommand.php]: Are you sure? If `symfony/symfony` requires `symfony/worker`, `symfony/symfony` should not be installable without filling all `symfony/worker`'s requirements, right?

Comment (lyrixx) [inline src/Symfony/Component/Worker/Command/WorkerRunCommand.php]: symfony/symfony does not require symfony/worker.

Comment (chalasr) [inline src/Symfony/Component/Worker/Command/WorkerRunCommand.php]: of course, sorry. re-added
